sigal/plugins/compress_assets.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+"""Plugin to compress static files for faster HTTP transfer.
+
+Currently, 3 methods are supported:
+    - gzip. No dependency required. This is the fastest, but also largest output.
+    - zopfli. Need zopfli module from https://pypi.python.org/pypi/zopfli. gzip compatible output with optimized size.
+    - brotli. Need brotli module from https://pypi.python.org/pypi/Brotli. Brotli is the best compressor for web usage.
+
+"""
+import logging
+import gzip
+import shutil
+import os
+import os.path
+
+from pathlib import Path
+
+from sigal import signals
+from click import progressbar
+
+logger = logging.getLogger(__name__)
+
+SETTINGS = {
+        'suffixes': ['htm', 'html', 'css', 'js', 'svg'],
+        'method': 'gzip',
+        }
+
+
+class BaseCompressor:
+
+    def __init__(self, settings):
+        self.settings = settings
+
+    def compressed_filename(self, filename):
+        suffix = self.suffix()
+        return filename.with_name(f'{filename.name}.{suffix}')
+
+    def suffix(self):
+        raise NotImplementedError
+
+    def do_compress(self, filename, compressed_filename):
+        raise NotImplementedError
+
+    def compress_file(self, filename):
+        compressed_filename = self.can_compress(filename)
+        if not compressed_filename:
+            return
+
+        self.do_compress(filename, compressed_filename)
+
+    def can_compress(self, filename):
+        # Be sure it's a path
+        filename = Path(filename)
+        if not filename.suffix[1:] in self.settings['suffixes']:
+            return False
+
+        file_stats = None
+        compressed_stats = None
+        compressed_filename_result = self.compressed_filename(filename)
+        try:
+            file_stats = os.stat(filename)
+            compressed_stats = os.stat(compressed_filename_result)
+        except FileNotFoundError:
+            pass
+
+        if file_stats and compressed_stats:
+            return compressed_filename_result if file_stats.st_mtime > compressed_stats.st_mtime else False
+        else:
+            return compressed_filename_result
+
+
+class GZipCompressor(BaseCompressor):
+
+    def suffix(self):
+        return 'gz'
```

Comment:
`suffix` could be a class variable instead, no need for a method.

Reply:
Yes indeed. I'll do it with my dev environment later, it's not a thing I can do without testing ;)